core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/NotificationEvent.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (C) 2018-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.core.events;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.ToString;
+
+import org.springframework.context.ApplicationEvent;
+import org.springframework.core.ResolvableType;
+import org.springframework.core.ResolvableTypeProvider;
+
+@Getter
+@ToString
+@EqualsAndHashCode
```

Comment:
Consider replacing these with @Data

Reply:
I tried to set `@Data`, but it doesn't compile.
The reason is that I'm extending an abstract class `org.springframework.context.ApplicationEvent` and `@Data` can't set in a subclass, It needs to be defined in the base class (due to `@RequiredArgsConstructor`) so I let only the ones required so it can be more _inmutable_ :)

Reply:
I just set `@Data` it compiled properly.
 
As note: It seems intellij has an error on `@Data` when it is used on subclasses showing ```lombok needs a default constructor in the base class```